jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/EtagUtil.java

```diff
@@ -0,0 +1,251 @@
+//
+
+package org.eclipse.jetty.http;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Objects;
+
+import org.eclipse.jetty.util.QuotedStringTokenizer;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.resource.Resource;
+
+/**
+ * Utility classes for Etag behaviors
+ */
+public class EtagUtil
+{
+    /**
+     * <p>The separator within an etag used to indicate a compressed variant.</p>
+     *
+     * <p>Default separator is {@code --}</p>
+     *
+     * <p>Adding a suffix of {@code gzip} to an etag of {@code W/"28c772d6"} will result in {@code W/"28c772d6--gzip"}</p>
+     *
+     * <p>The separator may be changed by the
+     * {@code org.eclipse.jetty.http.EtagUtil.ETAG_SEPARATOR} System property. If changed, it should be changed to a string
+     * that will not be found in a normal etag or at least is very unlikely to be a substring of a normal etag.</p>
+     */
+    public static final String ETAG_SEPARATOR = System.getProperty(EtagUtil.class.getName() + ".ETAG_SEPARATOR", "--");
+
+    /**
+     * Create a new {@link HttpField} {@link HttpHeader#ETAG} field suitable to represent the provided Resource.
+     *
+     * @param resource the resource to represent with an Etag.
+     * @return the field if possible to create etag, or null if not possible.
+     */
+    public static HttpField createWeakEtagField(Resource resource)
+    {
+        return createWeakEtagField(resource, null);
+    }
+
+    /**
+     * Create a new {@link HttpField} {@link HttpHeader#ETAG} field suitable to represent the provided Resource.
+     *
+     * @param resource the resource to represent with an Etag.
+     * @param etagSuffix the suffix for the etag
+     * @return the field if possible to create etag, or null if not possible.
+     */
+    public static HttpField createWeakEtagField(Resource resource, String etagSuffix)
+    {
+        Path path = resource.getPath();
+        if (path == null)
+            return null;
+
+        String etagValue = EtagUtil.calcWeakEtag(path, etagSuffix);
+        if (etagValue != null)
+            return new PreEncodedHttpField(HttpHeader.ETAG, etagValue);
+        return null;
+    }
+
+    /**
+     * Calculate the weak etag for the provided {@link Path} reference.
+     *
+     * <p>
+     * This supports relative path references as well.
+     * Which can be useful to establish a reliable etag if the base changes.
+     * </p>
+     *
+     * @param path the path to calculate from
+     * @return the weak etag
+     */
+    public static String calcWeakEtag(Path path)
+    {
+        return calcWeakEtag(path, null);
+    }
+
+    /**
+     * Calculate the weak etag for the provided {@link Path} reference.
+     *
+     * @param path the path to calculate from
+     * @param suffix the suffix for the ETag
+     * @return the weak etag
+     */
+    public static String calcWeakEtag(Path path, String suffix)
```

Comment:
Prefix `calc` implies calculations, which to me implies math operations only.
I think something like `compute` or `resolve` instead of `calc` is more appropriate.